src/commonMain/kotlin/exchange.dydx.abacus/processor/router/skip/SkipStatusProcessor.kt

```diff
@@ -0,0 +1,179 @@
+        IBCTransfer("ibc_transfer"),
+        AxelarTransfer("axelar_transfer"),
+        CCTPTransfer("cctp_transfer")
+    }
+
+    enum class TransferDirection(val value: String) {
+        From("from"),
+        To("to"),
+    }
+
+    data class ChainData(
+        val chainId: String? = null,
+        val transactionUrl: String? = null,
+        val txHash: String? = null,
+        val status: String? = null
+    ) {
+        fun toRouteStatusMap(): Map<String, String?> {
+            return mapOf(
+                "chainId" to chainId,
+                "txHash" to txHash,
+                "status" to status,
+            )
+        }
+        fun toTransferChainStatusMap(): Map<String, String?> {
+            return mapOf(
+                "transactionId" to txHash,
+                "transactionUrl" to transactionUrl,
+            )
+        }
+    }
+
+    private fun getStrFromMap(data: Map<String, Any?>, path: String): String? {
+        return parser.asString(parser.value(data, path))
+    }
+
+    private fun getSquidStatusFromState(state: String?): String? {
+        if (state == null) return null
+        if (state.contains("SUCCESS") || state.contains("RECEIVED")) return "success"
+        return "ongoing"
+    }
+
+    private fun getStatusFromTransferState(state: String?, transferDirection: TransferDirection): String? {
+//        CCTP transfer state:
+//        CCTP_TRANSFER_UNKNOWN - Unknown error
+//        CCTP_TRANSFER_SENT - The burn transaction on the source chain has executed
+//        CCTP_TRANSFER_PENDING_CONFIRMATION - CCTP transfer is pending confirmation by the cctp attestation api
+//        CCTP_TRANSFER_CONFIRMED - CCTP transfer has been confirmed by the cctp attestation api
+//        CCTP_TRANSFER_RECEIVED - CCTP transfer has been received at the destination chain
+
+//        Axelar transfer state:
+//        AXELAR_TRANSFER_UNKNOWN - Unknown error
+//        AXELAR_TRANSFER_PENDING_CONFIRMATION - Axelar transfer is pending confirmation
+//        AXELAR_TRANSFER_PENDING_RECEIPT - Axelar transfer is pending receipt at destination
+//        AXELAR_TRANSFER_SUCCESS - Axelar transfer succeeded and assets have been received
+//        AXELAR_TRANSFER_FAILURE - Axelar transfer failed
+
+//        IBC Transfer state:
+//        TRANSFER_UNKNOWN - Transfer state is not known.
+//        TRANSFER_PENDING - The send packet for the transfer has been committed and the transfer is pending.
+//        TRANSFER_RECEIVED - The transfer packet has been received by the destination chain. It can still fail and revert if it is part of a multi-hop PFM transfer.
+//        TRANSFER_SUCCESS - The transfer has been successfully completed and will not revert.
+//        TRANSFER_FAILURE - The transfer has failed.
+
+        if (state == null) return null
+//        If state is not unknown, it means the FROM tx succeeded
+        if (!state.contains("UNKNOWN") && transferDirection === TransferDirection.From) return "success"
```

Comment:
please prefer regular equality over referential equality. doesn't matter in this case since it's an enum, but generally referential equality is a smell and don't want to encourage it.

Reply:
ah TIL - thanks!